File: sharp_brain/sharp_brain/Neural/HiddenNeuron.cs
using SharpBrain.Activations;

namespace SharpBrain.Neural
{
    public class HiddenNeuron : WorkingNeuron
    {
        public HiddenNeuron(int layerIndex, int index, Activation activation)
            : base(layerIndex, index, activation)
        {

        }

        public void CalculateDelta(Layer nextLayer)
        {
            var derivative = activation.Derivative(netInput);
            var s = 0.0;
            foreach (var n in nextLayer)
            {
                var workingNeuron = n as WorkingNeuron;
                s += workingNeuron.Delta * workingNeuron.Connections[index].Weight;
            }
            delta = derivative * s;
        }
    }
}

File: kotlin_brians/src/main/kotlin/neurals/HiddenNeuron.kt
package neurals

import activations.Activation

class HiddenNeuron(layerIndex: Int, index: Int, activation: Activation) : WorkingNeuron(layerIndex, index, activation) {
    fun calculateDelta(nextLayer: List<WorkingNeuron>) {
        var s: Double = 0.0
        for (n in nextLayer) {
            s += n.delta * n.connections[index].weight
        }
        val derivative = activation.derivative(netInput)
        delta = s * derivative
    }
}